src/js/core.js

```diff
@@ -945,6 +949,37 @@ function MediumEditor(elements, options) {
                 }
             }
 
+            // If the selection is right at the ending edge of a link, put it outside the anchor tag instead of inside.
```

Comment:
I think that core code should not usually depend on extension code directly. I moved most of the code into a helper method now, so it should be easier to read.